TombEngine/Game/control/box.cpp

```diff
@@ -654,23 +637,30 @@ void CreatureUnderwater(ItemInfo* item, int depth)
 
 void CreatureFloat(short itemNumber) 
 {
-	auto* item = &g_Level.Items[itemNumber];
+	auto item = &g_Level.Items[itemNumber];
+	auto probe = GetCollision(item);
+	int floorHeight = probe.Position.Floor;
+	int ceilingHeight = probe.Position.Ceiling;
+
 	item->HitPoints = NOT_TARGETABLE;
 	item->Pose.Orientation.x = 0;
 
+	int y = item->Pose.Position.y;
+	if (y >= ceilingHeight - STEPUP_HEIGHT) // avoid the creature to go through the ceiling (if it happen !)
```

Comment:
What were the original countermeasures for creatures going through the ceiling? Or it is a fix for original issue?